asteroid/models/base_models.py

```diff
@@ -150,16 +167,22 @@ def file_separate(
                 est_src = resample(est_src, orig_sr=self.sample_rate, target_sr=fs)
             sf.write(save_name, est_src, fs)
 
-    def _separate(self, wav, *args, **kwargs):
-        """Hidden separation method
+    def forward_wav(self, wav, *args, **kwargs):
+        """Separation method for waveforms.
+        In case the network doesn't have waveforms as input/output, overwrite
+        this method to separate from waveform to waveform.
 
         Args:
             wav (Union[torch.Tensor, numpy.ndarray, str]): waveform array/tensor.
                 Shape: 1D, 2D or 3D tensor, time last.
-
-        Returns:
-            The output of self(wav, *args, **kwargs).
         """
+        if is_overridden("_separate", self, parent=BaseModel):
+            return self._separate(wav, *args, **kwargs)
+        return self(wav, *args, **kwargs)
```

Comment:
Ok, maybe the `is_overriden` is completely overkill.. 
Removing `_separate` from the base class and `hasattr(self,"_separate")` + raise warning if it does would have been enough. 
Hahah, I happy to write that function anyway :sweat_smile: